odoo/addons/base/models/res_partner.py

```diff
@@ -233,6 +233,13 @@ def _default_company(self):
         ('check_name', "CHECK( (type='contact' AND name IS NOT NULL) or (type!='contact') )", 'Contacts require a name.'),
     ]
 
+    @api.multi
+    def toggle_active(self):
+        for partner in self:
+            if partner.active and partner.user_ids:
+                raise ValidationError(_('You cannot archive a partner linked to internal user.'))
```

Comment:
Could we write in english? (grammar) ;-)
Also we tend to use more specific terminology than "partner" in the UI. Here it should probably be "contact".